app/src/main/java/com/criteo/testapp/MopubActivity.java

```diff
@@ -95,19 +95,24 @@ private void onBannerClick() {
     linearLayout.setVisibility(View.VISIBLE);
 
     publisherAdView = new MoPubView(this);
-    criteo.loadBid(BANNER, enrich((mThis, bid) -> mThis.criteo.enrichAdObjectWithBid(mThis.publisherAdView, bid)));
     publisherAdView.setAdUnitId(MOPUB_BANNER_ADUNIT_ID_HB);
-    publisherAdView.loadAd();
 
-    linearLayout.addView(publisherAdView);
+    criteo.loadBid(BANNER, enrich((mThis, bid) -> {
+      mThis.criteo.enrichAdObjectWithBid(mThis.publisherAdView, bid);
+
+      mThis.publisherAdView.loadAd();
+      mThis.linearLayout.addView(mThis.publisherAdView);
+    }));
   }
 
   private void onInterstitialClick() {
     MoPubInterstitial mInterstitial = new MoPubInterstitial(this, MOPUB_INTERSTITIAL_ADUNIT_ID_HB);
-    criteo.loadBid(INTERSTITIAL, enrich((mThis, bid) -> mThis.criteo.enrichAdObjectWithBid(mInterstitial, bid)));
-    mInterstitial.setInterstitialAdListener(
-        new TestAppMoPubInterstitialAdListener(TAG, mInterstitial));
-    mInterstitial.load();
+    mInterstitial.setInterstitialAdListener(new TestAppMoPubInterstitialAdListener(TAG, mInterstitial));
+
+    criteo.loadBid(INTERSTITIAL, enrich((mThis, bid) -> {
+      mThis.criteo.enrichAdObjectWithBid(mInterstitial, bid);
```

Comment:
nit/far: let's drop the hungarian notation :)

Reply:
the `mThis` is here to not leak `this`. Since it is not possible to name a variable `this` I choose `mThis` instead. `self` is also used in python but it's far from Java world.